source/php/Integrations/Image/ImageResolverInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace ComponentLibrary\Integrations\Image;
+
+/**
+ * Interface for resolving image URLs. 
+ * Used to resolve image URLs in the native system.
+ * 
+ * @package ComponentLibrary\Integrations\Image
+ */
+
+interface ImageResolverInterface
+{
+    /**
+     * Get the image URL by ID and size.
+     *
+     * @param int $id Image attachment ID.
+     * @param array $size Array containing image width and height.
+     * @return string|null URL of the image or null if not found.
+     */
+    public function getImageUrl(int $id, array $size): ?string;
```

Comment:
If $size needs to contain width and height. Could we not just allow $width and $height as params to the function instead?

Reply:
It is handled this way across the platform. I wanted to keep the same interface.